### DISCUSSION TITLE
Using o1 gives "Error connecting to server, try refreshing the page."

### DISCUSSION THREAD

**Discussion by AumCoin** [danny-avila/LibreChat#4280] [Troubleshooting] [answered]
I have the OpenAI Pay as you Go plan (probably lowest tier, if any) for which the pricing page seems to indicate I should have access to o1.  After updating Librechat and manually updating the list of models in .env with o1-preview and o1-mini, I try to use them but get this error.  All other OpenAI models work fine.  It takes not more than 10s for this message to appear after submitting a prompt.  Should I have access to the o1 models?

**Comment by danny-avila** [comment]
You're likely experiencing this: https://github.com/danny-avila/LibreChat/discussions/4111#discussioncomment-10696142

**Reply by AumCoin** [reply]
Sorry for being dense, I'm just using Docker and didn't add anything or do anything particularly "custom" (like adding an intermediate proxy).  Should that still apply to me?

**Reply by AumCoin** [reply]
Also fwiw I don't totally understand the user's comments about how 60s causes problems because it's via JS, but it seems like that probably doesn't apply to me anyway

**Reply by danny-avila** [reply]
A reverse proxy like NGINX will timeout connections taking long to respond, such as it would with o1.

If you don't have a service like that configured, then we need more information to debug this properly.

See if there's an error in the logs. If you are using docker, they are saved in the `./logs` directory at project root.

Check the latest error logs (files starting with "error" in name, i.e. `logs/error-2024-09-26.log`)

**Reply by AumCoin** [reply]
I see some 404 error messages indicating that "o1-mini" and "o1-preview" (with those names) do not exist or that I do not have access to them.  I had copied these from https://github.com/danny-avila/LibreChat/discussions/4037#discussioncomment-10678355 where the user said he had it working.  I had seen a comment somewhere to the effect that a user could "use the list provided by OpenAI", but when I omitted the env variable it did not show me any o1 models, so I tried adding them manually per this comment.

Can you tell me what the correct model strings to use are?

**Reply by fuegovic** [reply, marked as answer]
look here to make sure you have access to the "o1" models: https://platform.openai.com/settings/organization/limits
